cdap-hbase-compat-1.1/src/main/java/co/cask/cdap/data2/transaction/coprocessor/hbase11/DefaultTransactionProcessor.java

```diff
@@ -16,31 +16,136 @@
 import com.google.common.base.Supplier;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.CoprocessorEnvironment;
+import org.apache.hadoop.hbase.DoNotRetryIOException;
 import org.apache.hadoop.hbase.HTableDescriptor;
+import org.apache.hadoop.hbase.client.OperationWithAttributes;
 import org.apache.hadoop.hbase.coprocessor.RegionCoprocessorEnvironment;
 import org.apache.hadoop.hbase.filter.Filter;
 import org.apache.hadoop.hbase.regionserver.ScanType;
+import org.apache.hadoop.hbase.util.Bytes;
 import org.apache.tephra.Transaction;
+import org.apache.tephra.TxConstants;
 import org.apache.tephra.coprocessor.TransactionStateCache;
 import org.apache.tephra.hbase.coprocessor.CellSkipFilter;
 import org.apache.tephra.hbase.coprocessor.TransactionProcessor;
+import org.apache.tephra.util.TxUtils;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
 
 /**
  * Implementation of the {@link org.apache.tephra.hbase.coprocessor.TransactionProcessor}
  * coprocessor that uses {@link co.cask.cdap.data2.transaction.coprocessor.DefaultTransactionStateCache}
  * to automatically refresh transaction state.
  */
 public class DefaultTransactionProcessor extends TransactionProcessor {
+  private static final Log LOG = LogFactory.getLog(DefaultTransactionProcessor.class);
+
+  private CConfigurationCache cConfCache;
+  private String sysConfigTablePrefix;
+
+  @Override
+  public void start(CoprocessorEnvironment e) throws IOException {
+    if (e instanceof RegionCoprocessorEnvironment) {
+      RegionCoprocessorEnvironment env = (RegionCoprocessorEnvironment) e;
+      HTableDescriptor tableDesc = env.getRegion().getTableDesc();
+      String hbaseNamespacePrefix = tableDesc.getValue(Constants.Dataset.TABLE_PREFIX);
+
+      this.sysConfigTablePrefix = HTableNameConverter.getSysConfigTablePrefix(hbaseNamespacePrefix);
+      this.cConfCache = createCConfCache(env);
+    }
+    // Need to create the cConf cache before calling start on the parent, since it is required for
+    // initializing some properties in the parent class.
+    super.start(e);
+  }
+
+  @Override
+  public void stop(CoprocessorEnvironment e) throws IOException {
+    super.stop(e);
+    if (cConfCache != null) {
+      cConfCache.stop();
+    }
+  }
+
+  @Override
+  protected void ensureValidTxLifetime(RegionCoprocessorEnvironment env, OperationWithAttributes op,
+                                       @Nullable Transaction tx) throws IOException {
+    if (tx == null) {
+      return;
+    }
+
+    long maxLifetimeMillis;
+    if (txMaxLifetimeMillis == null) {
+      Configuration conf = getConfiguration(env);
+      if (conf != null) {
+        this.txMaxLifetimeMillis = TimeUnit.SECONDS.toMillis(conf.getInt(TxConstants.Manager.CFG_TX_MAX_LIFETIME,
```

Comment:
Looks like CDAP allows configuration to be loaded dynamically when CDAP is running. If so, we'll need to keep the same behavior here too.